app/services/IntersectService.ts

```diff
@@ -0,0 +1,35 @@
+import { checkIntersect as intersect } from '../helpers/Intersect';
+import { HealthcareAuthority } from '../store/types';
+import BackgroundTaskService from './BackgroundTaskService';
+
+class IntersectService {
+  isServiceRunning = false;
+  nextJob: HealthcareAuthority[] | null = null;
+
+  checkIntersect = (
+    healthcareAuthorities: HealthcareAuthority[] | null,
+    bypassTimer: boolean,
+  ): string => {
+    if (this.isServiceRunning) {
+      this.nextJob = healthcareAuthorities;
+      return 'skipped';
+    }
+    this.isServiceRunning = true;
+
+    intersect(healthcareAuthorities, bypassTimer).then(() => {
+      this.isServiceRunning = false;
+      if (this.nextJob) {
+        const job = this.nextJob;
+        this.nextJob = null;
```

Comment:
This looks to me like it will cancel lined up jobs?
I'm confused what the intention here is:
is this intended as a job queue, or to guarantee only one intersection search per HA is occurring?
I think it's the second for the case we're solving, so maybe we can tighten these invariants to only being a map of in progress one-off intersections?

My concern here i that this could be triggered for two separate valid HA lists and one would cancel the other

Reply:
I totally agree with you and a better approach for the intersection logic would be checking intersection per health authority and also keeping the exposure data per health authority, which will put us in a better spot for handling this issue.